homeassistant/components/ted5000/sensor.py

```diff
@@ -123,11 +269,50 @@ def update(self):
             doc = xmltodict.parse(request.text)
             mtus = int(doc["LiveData"]["System"]["NumberMTU"])
 
+            # MTU data
```

Comment:
Potentially this could be moved to an async context as the `get()` here will lock the thread until it completes.

Reply:
Spent a great amount of time trying to make it work as an async without success... Would you happen to know another XML scraping component that uses async that I could refer to?

As previously stated, I'm a very beginner in Python coding...

Reply:
I'm only adding readily available features to an existing component. I don't have the time/knowledge to redesign the component. I already fulfilled other requests, but for this one I tried but was unsuccessful.